test/fixtures/external-styles.js

```diff
@@ -0,0 +1,7 @@
+const foo = 'red'
+
+export default (
+  <style jsx>{`
+    p { color: ${foo} }
+  `}</style>
```

Comment:
This would be the best api however is not that easy to achieve this because in the consumer file (the one that imports this) we need to hash ``` `p { color: ${foo} }` ``` and once we have the hash we need to transform the css (add the prefix to it etc).